service/accessorFactory.go

```diff
@@ -38,6 +41,21 @@ func NewConsistentAccessorFactory(vnodeCount int) AccessorFactory {
 	}
 }
 
+// NewConsistentAccessorFactoryWithGate produces a factory which uses consistent hashing
+// of server nodes with the gate feature.  The returned factory does not modify instances passed to it.
+// Instances are hashed as is. If the gate is closed an error saying the gate is closed will be returned
+func NewConsistentAccessorFactoryWithGate(vnodeCount int, g gate.Interface) AccessorFactory {
```

Comment:
What's the use case for this?  Why doesn't the gate work as is via Alice-style decoration?